pkg/netpol/eval/internal/k8s/representative_selectors.go

```diff
@@ -0,0 +1,116 @@
+/*
+Copyright 2023- IBM Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package k8s
+
+import (
+	"crypto/sha1" //nolint:gosec // Non-crypto use
+	"encoding/hex"
+
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/selection"
+)
+
+// this file handles checking matching of representative peers' selectors' requirements
+// with rule selectors from policies
+// since a representative peer is a peer inferred from a network-policy rule
```

Comment:
```suggestion
// since a representative peer is a peer inferred from a network-policy rule
// for example, if a policy that captures a real workload `A` has a rule permitting egress connection, with selectors that match a representative peer `B`, 
// we need to capture this match, and thus infer the exposure connectivity from `A` to `B`
```